contrib/format-iceberg/src/main/java/org/apache/drill/exec/store/iceberg/plan/IcebergPluginImplementor.java

```diff
@@ -144,14 +141,18 @@ public boolean splitProject(Project project) {
 
   @Override
   public boolean canImplement(Project project) {
-    return true;
+    return hasIcebergGroupScan(project);
   }
 
   @Override
   public GroupScan getPhysicalOperator() {
     return groupScan;
   }
 
+  private boolean hasIcebergGroupScan(RelNode node) {
```

Comment:
It seems that this validation rule also applies to #2389?

Reply:
Yes, this PR also fixes the issue from that PR.